Mage.Sets/src/mage/cards/f/FromTheRubble.java

```diff
@@ -0,0 +1,58 @@
+package mage.cards.f;
+
+import java.util.UUID;
+
+import mage.abilities.Ability;
+import mage.abilities.Mode;
+import mage.abilities.common.AsEntersBattlefieldAbility;
+import mage.abilities.common.BeginningOfYourEndStepTriggeredAbility;
+import mage.abilities.effects.common.ChooseCreatureTypeEffect;
+import mage.abilities.effects.common.ReturnFromGraveyardToBattlefieldTargetEffect;
+import mage.abilities.effects.common.ReturnFromGraveyardToBattlefieldWithCounterTargetEffect;
+import mage.cards.CardImpl;
+import mage.cards.CardSetInfo;
+import mage.constants.CardType;
+import mage.constants.ComparisonType;
+import mage.constants.Outcome;
+import mage.counters.CounterType;
+import mage.filter.FilterCard;
+import mage.filter.common.FilterCreatureCard;
+import mage.filter.predicate.mageobject.ChosenSubtypePredicate;
+import mage.filter.predicate.mageobject.ManaValuePredicate;
+import mage.target.common.TargetCardInYourGraveyard;
+
+/**
+ *
+ * @author jimga150
+ */
+public final class FromTheRubble extends CardImpl {
+
+    public FromTheRubble(UUID ownerId, CardSetInfo setInfo) {
+        super(ownerId, setInfo, new CardType[]{CardType.ENCHANTMENT}, "{4}{W}{W}");
+        
+
+        // As From the Rubble enters the battlefield, choose a creature type.
+        this.addAbility(new AsEntersBattlefieldAbility(new ChooseCreatureTypeEffect(Outcome.PutCreatureInPlay)));
+
+        FilterCreatureCard filter = new FilterCreatureCard("a creature card of the chosen type");
```

Comment:
1. text should be "creature card of the chosen type from your graveyard" (since the target class doesn't add that text)
2. define the filter static, outside the constructor